src/TestSuite/TestCase.php

```diff
@@ -254,18 +242,61 @@ protected function setUp(): void
     protected function tearDown(): void
     {
         parent::tearDown();
+        $this->teardownFixtures();
+
         if ($this->_configure) {
             Configure::clear();
             Configure::write($this->_configure);
         }
         $this->getTableLocator()->clear();
         $this->_configure = [];
         $this->_tableLocator = null;
+    }
 
-        if ($this->fixtureManager) {
-            $this->fixtureManager->teardownTest($this);
-            $this->fixtureManager = null;
+    /**
+     * Initialized and loads any use fixtures.
+     *
+     * @return void
+     */
+    protected function setupFixtures(): void
+    {
+        $fixtureNames = $this->getFixtures();
+        if (empty($fixtureNames)) {
+            return;
+        }
+
+        if (static::$fixtureManager) {
+            // legacy fixtures are managed by FixtureInjector
+            if ($this->dropTables) {
+                deprecationWarning('`$dropTables` is deprecated and will be removed in 5.0.');
+            }
         }
+
+        $this->fixtureStrategy = $this->getFixtureStrategy();
+        $this->fixtureStrategy->setupTest($fixtureNames);
```

Comment:
Do we need the fixtures strategy if FixtureInjector is being used?

Reply:
The above check is missing a return - was there at some point.